### ISSUE TITLE
Special chars make navigation through search results inconsistent

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-server#6057]
Steps to reproduce:
1. Load AMO homepage https://addons.mozilla.org/en-US/firefox/
2. Type a special char in the search field i. e. "@" and make a search
3. Change filter menu option to "Any Firefox" and "All Systems"
4. Navigate to next page

Expected results:
Results are listed.

Actual results:
No results found.

Notes/Issues:
Verified on FF54(Win 7). Issue is reproducing on all AMO servers.
Video for this issue:
![no results](https://user-images.githubusercontent.com/15685960/28776265-940bd03c-75fe-11e7-8158-2c52954ebe81.gif)

**Comment by diox** [comment]
See also https://github.com/mozilla/addons-server/issues/3248

**Comment by EnTeQuAk** [comment]
Interesting, luckily this doesn't seem to be a regression. I can take a look to see wether this is ElasticSearch related or something else.

**Comment by EnTeQuAk** [comment]
ugh! I linked the commit to the wrong issue. reopening.

**Comment by EnTeQuAk** [comment]
Closing this together with landing https://github.com/mozilla/addons-server/pull/7303 since this improved the situation for special characters quite a bit in my experience. Please file a new issue if you find any more problems

**Comment by ioanarusiczki** [comment]
I verified this issue on AMO dev using FF 58 (Win10) and I reproduced this issue with the classic view. Still no results are found on the second page.  Also, when navigating back the first page will display layout issues.

![6057_diezvechi](https://user-images.githubusercontent.com/33448286/35566611-a81267d8-05ca-11e8-94e4-2e1db633188d.gif)


I also made some tests for the new frontend.  Sort by "Relevance" and Operation System "All" do not return any results.  "Any Firefox" criteria is not available here. 

![6057_dieznew](https://user-images.githubusercontent.com/33448286/35566619-ad70ead8-05ca-11e8-8e23-3ea233ec18be.gif)
